lib/gateway-kernel.coffee

```diff
@@ -0,0 +1,94 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+class GatewayKernel extends Kernel
+    constructor: (kernelSpec, session) ->
+        super kernelSpec
+
+        @session = session
+        @session.statusChanged.connect(@_onStatusChange, this)
```

Comment:
I checked again, and the code you suggested works.

Apparently the `TypeError`s I was getting before were unrelated to omitting the second argument.